msteams-platform/bots/how-to/format-ai-bot-messages.md

```diff
@@ -0,0 +1,459 @@
+
+:::image type="content" source="../../assets/images/bots/bot-feedback-buttons.png" border="false" alt-text="Screenshot shows the feedback buttons in a bot.":::
+
+When the user selects a feedback button, a respective feedback form appears based on the user's selection.
+
+# [👍Positive feedback](#tab/pos)
+
+:::image type="content" source="../../assets/images/bots/bot-feedback-form.png" border="false" alt-text="Screenshot shows the feedback form in a bot.":::
+
+# [👎Negative feedback](#tab/neg)
+
+*Image placeholder*
+
+---
+
+### Enable feedback buttons
+
+To enable feedback buttons in your bot, add a new `channelData` object in your bot's message and set `feedbackLoopEnabled` to true.
+
+The following code snippet shows how to enable feedback buttons in a bot:
+
+```json
+{
+    "type": "message",
+    "from": {
+        "id": "28:48b2e1fd-b6b8-46ce-8074-8e918304fcb2"
+    },
+    "conversation": {
+        "isGroup": true,
+        "id": "19:21e795db-9804-4bb8-9a6a-f85b9e8e9602@thread.tacv2"
+    },
+    "recipient": {
+        "id": "8:orgid:67cc58d0-4a80-496a-a019-437b70bc6529"
+    },
+    "text": "93001cd5-45f4-4a9a-b694-0f44eecfd004",
+    "attachments": [],
+    "entities": [],
+    "channelData": {
+        "feedbackLoopEnabled": true
+    }
+}
+
+```
+
+For a bot built using Teams AI library, Teams enables feedback buttons to all bot messages when `enable_feedback_loop` is set to `true`.
+
+[Sample code reference](https://github.com/microsoft/teams-ai/blob/main/js/samples/04.ai-apps/h.datasource-azureOpenAI/src/app.ts)
+
+```javascript
+await context.sendActivity({
+  type: ActivityTypes.Message,
+  text: `Hey I'm a friendly AI bot. This message is generated via AI - ${txt}`,
+  channelData: {
+    feedbackLoopEnabled: true // Enable feedback buttons
+  },
+});
+```
+
+After you enable feedback buttons, the footer of your bot's message contains a like and dislike button for the user to select. You can collect feedback on your bot’s responses from personal chats, group chats, and channels.
+
+The bot sends the user's input, received in the feedback form, through a bot invoke. The following code snippet shows a bot invoke containing positive feedback from a user:
+
+```json
+{
+ "name": "message/submitAction",
+ "conversation": {
+  "id": "19:144de878089f94d7db699d10a4672bc040@thread.v2"
+ },
+ "value": {
+  "actionName": "feedback",
+  "actionValue": {
+    "reaction": "like",
+    "feedback": "test feedback"
+  }
+ }
+}
+```
+
+---
+
+### Handle feedback
+
+If your bot is built using Teams AI library, the feedback received is automatically handled. For other types of bots, you need to have an `onInvokeActivity` handler to correctly process the invoke when the bot receives it. Ensure that you return a status code 200 with an empty response.
```

Comment:
This is incorrect. https://github.com/microsoft/teams-ai/blob/main/getting-started/CONCEPTS/POWERED-BY-AI.md#feedback-loop

Reply:
@corinagum, we've updated this section. Could you take a look and let us know if it reads better now?

Reply:
@v-kushals This looks good to me, thanks for making the changes